internal/encryption/keyprovider/output.go

```diff
@@ -0,0 +1,24 @@
+package keyprovider
+
+import "github.com/zclconf/go-cty/cty"
+
+type Output struct {
+	EncryptionKey []byte `hcl:"encryption_key" cty:"encryption_key" json:"encryption_key" yaml:"encryption_key"`
+	DecryptionKey []byte `hcl:"decryption_key" cty:"decryption_key" json:"decryption_key" yaml:"decryption_key"`
+	Metadata      any
+}
```

Comment:
```suggestion
// Output is the standardized structure a key provider must return when providing a key.
// It contains two keys because some key providers may prefer include random data (e.g. salt)
// in the generated keys. 
// Additionally, the Metadata should contain a struct with `meta` tags that OpenTofu can serialize
// into the encrypted form. OpenTofu will inject the `meta` tagged items back into the
// key provider configuration when an item needs to be decrypted.
type Output struct {
	EncryptionKey []byte `hcl:"encryption_key" cty:"encryption_key" json:"encryption_key" yaml:"encryption_key"`
	DecryptionKey []byte `hcl:"decryption_key" cty:"decryption_key" json:"decryption_key" yaml:"decryption_key"`
	Metadata      any
}
```